pkg/sql/testdata/rename_index

```diff
@@ -74,3 +74,12 @@ SELECT * FROM users
 ----
 1 tom   cat
 2 jerry rat
+
+statement ok
+CREATE VIEW v AS SELECT name FROM users@{FORCE_INDEX=ufo}
+
+statement error cannot rename index "ufo" because it is depended on by view "test.v"
```

Comment:
No, I wasn't suggesting the rewriting was easy, just wondering what the long-term goal is here. This is fine in the short-term.

One thought about how to avoid the VIEW query rewriting is that we can extend our SQL syntax in order to allow referring to tables/columns/indexes by ID. Then we'd do the name resolution once when the VIEW is created. Easy to describe this, but it would likely be a whole bunch of work to actually accomplish.